src/jit/flowgraph.cpp

```diff
@@ -21588,6 +21587,26 @@ Compiler::fgWalkResult      Compiler::fgUpdateInlineReturnExpressionPlaceHolder(
         }
         while (tree->gtOper == GT_RET_EXPR);
 
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+        if (tree->gtOper == GT_CALL && tree->AsCall()->HasMultiRegRetVal())
+        {
+            // For var = call() with long return types, we need to set
+            // lvIsMultiRegArgOrRet so we do not promote the var.
+            GenTreePtr parent = data->parent;
+            if (parent->gtOper == GT_ASG)
+            {
+                GenTree* op1 = parent->gtOp.gtOp1;
+                if (op1->gtOper == GT_LCL_VAR)
+                {
+                    GenTreeLclVarCommon* varNode = op1->AsLclVarCommon();
+                    unsigned varNum = varNode->gtLclNum;
+                    assert(varNum < comp->lvaCount);
+                    comp->lvaTable[varNum].lvIsMultiRegArgOrRet = true;
+                }
+            }
+        }
+#endif
+
```

Comment:
OK, after our discussion, it is clear that we probably need to not just check for `op1->gtOper == GT_LCL_VAR`, but ensure that it is. This would probably be done in an analog to `Compiler::fgAssignStructInlineeToVar()`.